src/commands.ts

```diff
@@ -97,125 +111,163 @@ export class Commands {
         placeHolder: "Copy your API key from the opened browser page.",
         value: await this.storage.getSessionToken(),
         ignoreFocusOut: true,
-        validateInput: (value) => {
-          return axios
-            .get("/api/v2/users/me", {
-              baseURL: url,
-              headers: {
-                "Coder-Session-Token": value,
-              },
-            })
-            .then(() => {
-              return undefined
-            })
-            .catch((err) => {
-              if (err instanceof CertificateError) {
-                err.showNotification()
-
-                return {
-                  message: err.x509Err || err.message,
-                  severity: vscode.InputBoxValidationSeverity.Error,
-                }
-              }
-              // This could be something like the header command erroring or an
-              // invalid session token.
-              const message =
-                err?.response?.data?.detail || err?.message || err?.response?.status || "no response from the server"
+        validateInput: async (value) => {
+          this.restClient.setSessionToken(value)
```

Comment:
If `validateInput` fails, do you think it makes sense to set the token back to the previous value?

Reply:
I was thinking about this too haha and decided to be lazy

But yeah you are right, if this fails I should reset the token and probably the URL as well.

Would be cool if I could set it for just one request.

Actually maybe I should just create a new client just for this validation?  Then once it validates, then I set it on the main client.  Yeah I think that makes sense.